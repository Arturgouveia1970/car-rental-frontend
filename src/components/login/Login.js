import React, { useState } from 'react';
import { NavLink, Navigate } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { login } from '../../Redux/User/registerSlice';
import styles from './Login.module.css';

/* eslint-disable jsx-a11y/label-has-associated-control */
const Login = () => {
  const [email, setEmail] = useState('');
  // const [password, setPassword] = useState('');
  const [loading, setLoading] = useState(false);
  const dispatch = useDispatch();
  const user = useSelector((state) => state.user);

  const formSubmit = (e) => {
    e.preventDefault();
    if (email) {
      setLoading(true);
      dispatch(login({ email, setLoading }));
    }
  };

  if (user.logged_in) {
    return <Navigate replace to="/home" />;
  }

  return (
    <div
      className={`${styles.container} container-fluid vw-100 d-flex flex-column align-items-center justify-content-center`}
    >
      <h2 className={`${styles.header} fs-2 mb-5`}>Log in</h2>
      <form onSubmit={formSubmit} className={styles.form}>
        <div className="form-floating mb-3">
          <input
            type="text"
            className="form-control"
            id="floatingUsername"
            placeholder="Email"
            value={email}
            required
            onChange={(e) => setEmail(e.target.value)}
          />
          <label htmlFor="floatingUsername">email address</label>
        </div>

        {/* <div className="form-floating mb-3">
          <input
            type="text"
            className="form-control"
            id="floatingUsername"
            placeholder="Password"
            value={password}
            required
            onChange={(e) => setPassword(e.target.value)}
          />
          <label htmlFor="floatingUsername">Password</label>
        </div> */}
        {loading ? (
          <button type="button" className="btn btn-primary disabled mb-3">
            <i className="fa-solid fa-spinner fa-spin" />
          </button>
        ) : (
          <button type="submit" className="btn btn-primary mb-3">
            Log in
          </button>
        )}
      </form>
      <p className={`${styles.errorMsg} fs-5`}>{user.error}</p>
      <NavLink className={styles.button} to="/register">
        <span className="btn btn-info mb-0">Sign up</span>
      </NavLink>
    </div>
  );
};

export default Login;
