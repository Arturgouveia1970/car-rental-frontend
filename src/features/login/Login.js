// eslint camelcase: ["error", {properties: "never"}]

import React, { useState } from 'react';
import { NavLink, Navigate } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { login } from '../../Redux/Users/registerSlice';
import styles from './Login.module.css';

/* eslint-disable jsx-a11y/label-has-associated-control */
const Login = () => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [loading, setLoading] = useState(false);
  const dispatch = useDispatch();
  const [logged_in, setLogged_in] = useState(false);
  const user = useSelector((state) => state.user);

  const formSubmit = (e) => {
    e.preventDefault();
    if (password && email) {
      setLoading(true);
      setLogged_in(true);
      dispatch(login({
        email, password, setLoading, setLogged_in,
      }));
    }
  };

  if (logged_in) {
    return <Navigate to="/home" />;
  }

  return (
    <div
      className={`${styles.container}
        container-fluid vw-100 d-flex flex-column align-items-center justify-content-center`}
    >
      <h2 className={`${styles.header} fs-2 mb-5`}>Log in</h2>
      <form onSubmit={formSubmit} className={styles.form}>
        <div className="form-floating mb-3">
          <input
            type="email"
            className="form-control"
            id="floatingInput"
            placeholder="name@example.com"
            value={email}
            required
            onChange={(e) => setEmail(e.target.value)}
          />
          <label htmlFor="floatingInput">Email address</label>
        </div>
        <div className="">
          <input
            type="password"
            className="form-control"
            id="floatingInput"
            placeholder="password"
            value={password}
            required
            onChange={(e) => setPassword(e.target.value)}
          />
          <label htmlFor="floatingInput"> </label>
        </div>
        {loading ? (
          <button type="button" className="btn btn-primary disabled mb-3">
            <i className="fa-solid fa-spinner fa-spin" />
          </button>
        ) : (
          <button type="submit" className="btn btn-primary mb-3">
            Log in
          </button>
        )}
      </form>
      <p className={`${styles.errorMsg} fs-5`}>{user}</p>
      <NavLink to="/register">
        <span className="btn btn-secondary">Sign up</span>
      </NavLink>
    </div>
  );
};

export default Login;

// /* eslint linebreak-style: ["error", "windows"] */
// import { Link, useParams } from 'react-router-dom';
// import { useState, useEffect } from 'react';
// import { useDispatch } from 'react-redux';
// import axios from 'axios';
// // import Button from '../components/button/Button';
// // import './Login.scss';

// const LoginScreen = () => {
//   const { id } = useParams();
//   const [email, setEmail] = useState();
//   const [password, setPassword] = useState();
//   const [loading, setLoading] = useState(false);

//   // const [isActive, setActive] = useState(true);

//   // const ToggleClass = () => {
//   //   setActive(false);
//   // };
//   const dispatch = useDispatch();

//   const fetchUser = (id) => (async (dispatch) => {
//     setLoading(true);
//     // await axios get function
//     await axios
//       .get(`https://dreamcars.onrender.com/api/v1/users/${id}`)
//       .then((response) => {
//         setEmail(response.data);
//         setPassword(response.data);
//         // console.log(response.data);
//         setLoading(false);
//         dispatch({ type: 'FETCH_USERS', payload: response });
//       });
//   });

//   useEffect(() => {
//     fetchUser(id);
//     dispatch(id);
//   }, [id, fetchUser, dispatch]);

//   if (loading) {
//     return (
//   <div className="container-fluid vh-100 v-100 d-flex justify-content-center align-items-center">
//         <i className="fa-solid fa-spinner fa-spin fs-1" />
//       </div>
//     );
//   }
//   return (
//     <div className="container page-login">

//       <form
//         action="Post"
//         className="login-form container-fluid vw-100 d-flex flex-column
//           align-items-center justify-content-center" method="POST" ref={fetchUser}>
//         <h2>LOGIN</h2>

//         {/* <div className="add-padding-below">
//           <span
//             className={
//             isActive ? 'error-message' : 'error-message active'
//             }
//           >
//             Error with credentials
//           </span>
//         </div> */}

//         <div className="add-padding-below">
//           <input
//             value={email}
//             type="text"
//             id="email"
//             name="email"
//             className="form-field"
//             placeholder="example@email.com"
//             required
//           />
//         </div>
//         <div className="add-padding-below">
//           <input
//             value={password}
//             type="text"
//             id="password"
//             name="password"
//             className="form-field"
//             placeholder="Password"
//             required
//           />
//         </div>

//         <p className="signin-message">
//           Don&apos;t have an account yet?
//           <Link to="/register">
//             SIGN UP
//           </Link>
//         </p>

//         <Link to={`/home/${id}`} className="form-bottom-bar">
//           <button
//             type="submit"
//             className="btn btn-primary mb-3"
//             onClick={fetchUser}
//           >
//             Log in
//           </button>
//         </Link>
//       </form>

//     </div>
//   );
// };

// export default LoginScreen;
