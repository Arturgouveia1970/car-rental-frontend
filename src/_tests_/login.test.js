import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Login from '../components/login/Login';
import store from '../Redux/configureStore';

describe('Log in tests', () => {
  it('renders correctly', () => {
    const tree = render(
      <Provider store={store}>
        <BrowserRouter>
          <Login />
        </BrowserRouter>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Log in page has log in button', () => {
    const login = render(
      <>
        <Provider store={store}>
          <BrowserRouter>
            <Login />
          </BrowserRouter>
        </Provider>
      </>,
    );

    expect(login.findByText('Log in')).toBeTruthy();
  });

  it('Log in page has a sign up link', () => {
    const login = render(
      <>
        <Provider store={store}>
          <BrowserRouter>
            <Login />
          </BrowserRouter>
        </Provider>
      </>,
    );

    expect(login.findByText('Sign up')).toBeTruthy();
  });
});
