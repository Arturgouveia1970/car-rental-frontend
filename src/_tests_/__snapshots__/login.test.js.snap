// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Log in tests renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container container-fluid vw-100 d-flex flex-column align-items-center justify-content-center"
      >
        <h2
          class="header fs-2 mb-5"
        >
          Log in
        </h2>
        <form
          class="form"
        >
          <div
            class="form-floating mb-3"
          >
            <input
              class="form-control"
              id="floatingUsername"
              placeholder="Email"
              required=""
              type="text"
              value=""
            />
            <label
              for="floatingUsername"
            >
              email address
            </label>
          </div>
          <button
            class="btn btn-primary mb-3"
            type="submit"
          >
            Log in
          </button>
        </form>
        <p
          class="errorMsg fs-5"
        />
        <a
          class="button"
          href="/register"
        >
          <span
            class="btn btn-info mb-0"
          >
            Sign up
          </span>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container container-fluid vw-100 d-flex flex-column align-items-center justify-content-center"
    >
      <h2
        class="header fs-2 mb-5"
      >
        Log in
      </h2>
      <form
        class="form"
      >
        <div
          class="form-floating mb-3"
        >
          <input
            class="form-control"
            id="floatingUsername"
            placeholder="Email"
            required=""
            type="text"
            value=""
          />
          <label
            for="floatingUsername"
          >
            email address
          </label>
        </div>
        <button
          class="btn btn-primary mb-3"
          type="submit"
        >
          Log in
        </button>
      </form>
      <p
        class="errorMsg fs-5"
      />
      <a
        class="button"
        href="/register"
      >
        <span
          class="btn btn-info mb-0"
        >
          Sign up
        </span>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
